src/Microsoft.ML.Tokenizers/Model/Tiktoken.cs

```diff
@@ -784,14 +804,14 @@ internal static (Dictionary<string, int> SpecialTokens, Regex Regex, string Url)
                 }
             }
 
-            if (!_tiktokenCache.TryGetValue(mergeableRanksFileUrl, out (Dictionary<ReadOnlyMemory<byte>, int> encoder, Dictionary<StringSpanOrdinalKey, int> vocab, Dictionary<int, ReadOnlyMemory<byte>> decoder) cache))
+            if (!_tiktokenCache.TryGetValue(mergeableRanksFile, out (Dictionary<ReadOnlyMemory<byte>, int> encoder, Dictionary<StringSpanOrdinalKey, int> vocab, Dictionary<int, ReadOnlyMemory<byte>> decoder) cache))
             {
-                using (Stream stream = await Helpers.GetStreamAsync(_httpClient, mergeableRanksFileUrl, cancellationToken).ConfigureAwait(false))
-                {
-                    cache = await LoadTikTokenBpeAsync(stream, useAsync: true, cancellationToken).ConfigureAwait(false);
-                }
+                using Stream compressedStream = typeof(Tokenizer).Assembly.GetManifestResourceStream(mergeableRanksFile)!;
+                using Stream deflateStream = new DeflateStream(compressedStream, CompressionMode.Decompress);
+
+                cache = await LoadTikTokenBpeAsync(deflateStream, useAsync: true, cancellationToken).ConfigureAwait(false);
```

Comment:
This is no longer doing actual I/O: GetManifestResourceStream just returns an UnmanagedResourceStream, so all the reading this does is against in-memory data and will all complete synchronously.

As such this no longer needs to be async or Task returning.

Reply:
ok, will change that.

Reply:
I think we don't need to change LoadTikTokenBpeAsync. The reason is this we still support 

```C#
        public static async Task<Tiktoken> CreateAsync(
                            Stream vocabStream,
                            IReadOnlyDictionary<string, int>? specialTokens = null,
                            int cacheSize = LruCache<int[]>.DefaultCacheSize,
                            CancellationToken cancellationToken = default)
```

We may need to change only the call sites which using the embedding stream.

Reply:
> The reason is this we still support

Ah, ok, if the public API allows a Stream to be passed in, makes sense.

Reply:
I have removed the public API 

```C#
        public static Task<Tokenizer> CreateTiktokenForModelAsync(
                                                string modelName,
                                                IReadOnlyDictionary<string, int>? extraSpecialTokens = null,
                                                Normalizer? normalizer = null,
                                                CancellationToken cancellationToken = default)
```

There is no async operation executed in the method after we read the stream from the embedding files. We already have `CreateTiktokenForModel` too.